File: stack/python/stack_array_implementation.py
from stack_interface import StackInterface, StackSuper

class StackArrayImpl(StackSuper):
    def __init__(self, stack_size: int) -> None:
        self.stack = []
        self.stack_size = stack_size

    def pop(self) -> any:
        if self.size() > 0:
            return self.stack.pop()
        return None
    
    def push(self, item: any) -> None:
        if not self.is_full():
            self.stack.append(item)

    def size(self) -> int:
        return len(self.stack)
    
    def is_full(self) -> bool:
        return self.size() >= self.stack_size
    
    def is_empty(self) -> bool:
        return self.size() == 0
    
    def peek(self) -> any:
        if self.size() > 0:
            return self.stack[self.size() - 1]
        return None

File: stack/java/Stack/src/main/java/org/example/StackArrayImplementation.java
package org.example;

public class StackArrayImplementation<E> implements StackInterface<E>{
    private E[] stack;
    private int currentSize = 0;

    @SuppressWarnings("unchecked")
    StackArrayImplementation(int maxSize){
        this.stack = (E[]) new Object[maxSize];
    }

    public E peek(){
        if (!isEmpty()){
            return stack[currentSize - 1];
        }
        return null;
    }
    public void push(E item){
        if (!isFull()){
            stack[size()] = item;
            currentSize++;
        }
    }
    public E pop(){
        if (!isEmpty()){
            E poppedItem = stack[currentSize - 1];
            stack[currentSize - 1] = null;
            currentSize--;
            return poppedItem;
        }

        return null;
    }
    public int size(){
        return currentSize;
    }
    public boolean isEmpty(){
        return currentSize < 1;
    }
    public boolean isFull(){
        return currentSize >= stack.length;
    }
}
